### ISSUE TITLE
Understanding the 'no-transformation' case

### ISSUE THREAD

**Issue by MKowal2** [xu-ji/IIC#43]
I just had a question about the case where one uses NO transformation g for image clustering (i.e. x = x'). I am playing around with the mutual information loss and it appears to be learning something when trained without transformations. For starters, it learns to predict a one-hot distribution to minimize H(z|z'), but why then is it able to get larger than random chance on the validation set? (10 classes, 18% accurate after a few epochs)

Could anyone help me understand why the network learns something when we maximize the mutual information between an image and the exact same image? Or is it more likely that there is a bug in my code? Thanks! and great work :)

**Comment by xu-ji** [comment]
Actually we did mention this in the original paper. On page 4, 

> For example, even for the degenerate case of identical pairs x = x', the IIC objective encourages a deterministic clustering function (i.e. \Phi(x) is a one-hot vector) as this results in null conditional entropy H(z|z') = 0.

Even in the case where the pairs contain the same image, the entropy maximisation will still push for even cluster size and the conditional entropy minimisation will still push for one-hot predictions (cluster assignment being a discrete variable). CNNs are biased towards making the same prediction for different images that are close in pixel space. These factors alone are enough to improve predictive behaviour w.r.t. random initialisation.

But we already know that performance will not be as good as if you include invariance to transforms. Because the imposition of invariance to non-material changes is an additional optimisation constraint that guides the learned prediction behaviour closer to what we know it should be like.

**Comment by MKowal2** [comment]
Yea I saw the explanation in the paper, but your response is very helpful. 

Even cluster size and one hot predictions for images close together in pixel space due to CNN bias. After training the model without transformations, the accuracy is less than those with transformations. 
Thank you for the response!
